tests/pytests/test_info_modules.py

```diff
@@ -899,6 +982,108 @@ def test_args_errors_cluster(self):
+
+    # ---------- Timeout Warnings ----------
+    allShards_change_timeout_policy(self.env, 'RETURN')
+
+    coord_before_warn = info_modules_to_dict(self.env)
+    shards_before_warn = {i: info_modules_to_dict(self.env.getConnection(i)) for i in range(1, self.env.shardsCount + 1)}
+    base_warn_coord = int(coord_before_warn[COORD_WARN_ERR_SECTION][TIMEOUT_WARNING_COORD_METRIC])
+    base_warn_shards = {i: int(shards_before_warn[i][WARN_ERR_SECTION][TIMEOUT_WARNING_SHARD_METRIC]) for i in shards_before_warn}
+
+    # Test timeout warning in FT.SEARCH (shards)
+    self.env.expect(debug_cmd(), 'FT.SEARCH', 'idx', '*',
+                    'TIMEOUT_AFTER_N', 0, 'DEBUG_PARAMS_COUNT', 2).noError()
+    # Shards: +1 each
+    for shardId in range(1, self.env.shardsCount + 1):
+      info_dict = info_modules_to_dict(self.env.getConnection(shardId))
+      self.env.assertEqual(info_dict[WARN_ERR_SECTION][TIMEOUT_WARNING_SHARD_METRIC], str(base_warn_shards[shardId] + 1),
+                           message=f"Shard {shardId} SEARCH timeout warning should be +1")
+    # Coord: unchanged
+    info_coord = info_modules_to_dict(self.env)
+    self.env.assertEqual(info_coord[COORD_WARN_ERR_SECTION][TIMEOUT_WARNING_COORD_METRIC], str(base_warn_coord),
+                         message="Coordinator timeout warning should not change after FT.SEARCH")
+
+    # Test timeout warning in FT.AGGREGATE (shards only via INTERNAL_ONLY)
+    self.env.expect(debug_cmd(), 'FT.AGGREGATE', 'idx', '*',
+                    'TIMEOUT_AFTER_N', 1, 'INTERNAL_ONLY', 'DEBUG_PARAMS_COUNT', 3).noError()
+    # Shards: +1 each again (total +2)
+    for shardId in range(1, self.env.shardsCount + 1):
+      info_dict = info_modules_to_dict(self.env.getConnection(shardId))
+      self.env.assertEqual(info_dict[WARN_ERR_SECTION][TIMEOUT_WARNING_SHARD_METRIC], str(base_warn_shards[shardId] + 2),
+                           message=f"Shard {shardId} AGG INTERNAL_ONLY timeout warning should be +2 total")
+    # Coord: unchanged
+    info_coord = info_modules_to_dict(self.env)
+    self.env.assertEqual(info_coord[COORD_WARN_ERR_SECTION][TIMEOUT_WARNING_COORD_METRIC], str(base_warn_coord),
+                         message="Coordinator timeout warning should not change after AGG INTERNAL_ONLY")
+
+    # Test timeout warning in FT.AGGREGATE (coordinator)
+    self.env.expect(debug_cmd(), 'FT.AGGREGATE', 'idx', '*',
+                    'TIMEOUT_AFTER_N', 0, 'DEBUG_PARAMS_COUNT', 2).noError()
+    # Shards: +3 (timeout is returned by the coord, but each shard still times out)
+    for shardId in range(1, self.env.shardsCount + 1):
+      info_dict = info_modules_to_dict(self.env.getConnection(shardId))
+      self.env.assertEqual(info_dict[WARN_ERR_SECTION][TIMEOUT_WARNING_SHARD_METRIC], str(base_warn_shards[shardId] + 3),
+                           message=f"Shard {shardId} AGG coordinator timeout warning should be +3 total")
```

Comment:
This one can be flaky
the coordinator might return before the shards reach to the point where they update the timeout wraning